### PR TITLE
Pseudonymizer stricter schema

### PR DESCRIPTION
**Pull request by haraldsvik**
- Cast values to string before sending it to pseudonymizer.
- Set specific type when writing to parquet again
- Added tests to verify parquet column types are as expected

### CODE REVIEW

File: tests/unit/worker/steps/test_dataset_pseudonymizer.py [hunk 1]
```diff
@@ -186,3 +303,21 @@ def test_pseudonymizer_invalid_unit_id_type():
     with pytest.raises(BuilderStepError) as e:
         dataset_pseudonymizer.run(INPUT_PARQUET_PATH, INVALID_METADATA, JOB_ID)
     assert f"Failed to pseudonymize, UnregisteredUnitType: {str(e)}"
+
+
+# In Parquet, the physical type refers to how the data is stored.
```
Comment (DanielElisenberg) [inline tests/unit/worker/steps/test_dataset_pseudonymizer.py:261]: TIL :100:

File: job_executor/worker/steps/dataset_pseudonymizer.py [hunk 2]
```diff
@@ -90,9 +99,21 @@ def _pseudonymize(
...
             continue
-        columns.append(
-            input_dataset.to_table(columns=[column_name])[column_name]
-        )
+        if column_name in [
+            "start_year",
+            "start_epoch_days",
+            "stop_epoch_days",
+        ]:
+            columns.append(
```
[the remark this answers is not in the document]
Comment (haraldsvik) [reply job_executor/worker/steps/dataset_pseudonymizer.py:107]: ah then it should be string!

File: job_executor/worker/steps/dataset_pseudonymizer.py [hunk 3]
```diff
@@ -45,14 +45,23 @@ def _pseudonymize_column(
         identifiers_table[column_name]
     ).to_pylist()
 
+    unique_identifiers = [str(item) for item in unique_identifiers]
```
[the remark this answers is not in the document]
Comment (haraldsvik) [reply job_executor/worker/steps/dataset_pseudonymizer.py:48]: Yes keeping it in pyarrow will probably make it faster too!
if im reading the pyarrow docs correctly "cast" does not do anything if the source and the target type is the same. So should not lose much performance if the indentifiers_table is already a string.
```py
    string_identifiers = identifiers_table[column_name].cast(pyarrow.string())
    unique_identifiers = compute.unique(string_identifiers).to_pylist()
  ```

**Review by DanielElisenberg — APPROVED** [review]
:100:

**Pull request merged**